src/codegen/struct_layout.rs

```diff
@@ -262,6 +268,27 @@ impl<'a> StructLayoutTracker<'a> {
         padding_layout.map(|layout| self.padding_field(layout))
     }
 
+    pub fn add_tail_padding(
+        &mut self,
+        comp_name: &str,
+        comp_layout: Layout,
+    ) -> Option<proc_macro2::TokenStream> {
```

Comment:
I'd prefer to write this as:

```
if !self.ctx.options().force_explicit_padding {
    return None;
}

if self.latest_offset != comp_layout.size {
    return None;
}

...
```

But no big deal either way.

Reply:
That style is fine with me. I pushed an updated version with this change.